src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx

```diff
@@ -21,84 +21,56 @@ type SelectInputProps = {
 
-        this.state = {
-            value: defaultOption.displayValue,
-            showDropdown: false,
-        };
+    if (defaultOption === undefined) {
+        defaultOption = { value: 0, displayValue: '' };
     }
 
-    private onClicked = () => {
-        this.setState((prevState) => {
-            const newState = { ...prevState };
-
-            newState.showDropdown = !prevState.showDropdown;
+    const [value, setValue] = useState<any>(defaultOption.displayValue);
+    const [showDropdown, setShowDropdown] = useState(false);
 
-            return newState;
-        });
-    }
-
-    private onOptionClicked = (option: Option) => {
-        if (this.props.onChange) {
-            this.props.onChange(option.value);
+    const onOptionClicked = (option: Option) => {
+        if (props.onChange) {
+            props.onChange(option.value);
         }
+        setValue(option.displayValue);
+    };
 
-        this.setState((prevState) => {
-            const newState = { ...prevState };
-
-            newState.value = option.displayValue;
-
-            return newState;
-        });
-    }
-
-    private dropdownElements = (): JSX.Element[] => {
+    const dropdownElements = (): JSX.Element[] => {
         const optionElements: JSX.Element[] = [];
 
-        for (const option of this.props.options) {
+        for (const option of props.options) {
             optionElements.push((
-                <div className="text-white hover:bg-white hover:bg-opacity-5 transition duration-300 rounded-lg px-5 py-1.5" onClick={() => this.onOptionClicked(option)}>
+                <div className="text-white hover:bg-white hover:bg-opacity-5 transition duration-300 rounded-lg px-5 py-1.5" onClick={() => onOptionClicked(option)}>
                     {option.displayValue}
                 </div>
             ));
         }
         return optionElements;
-    }
-
-    render() {
-        return (
-            <div className={`flex ${this.props.reverse ? 'flex-row-reverse' : 'flex-row'}`}>
-                <div className={`text-lg flex flex-grow m-2.5 items-center ${this.props.reverse ? 'justify-start' : 'justify-end'}`}>{this.props.label}</div>
-                <div className="flex items-center cursor-pointer relative" onClick={this.onClicked}>
-                    <div className={`relative flex px-5 py-1.5 text-lg text-white rounded-lg bg-navy-light border-2 border-navy-light
-                        focus-within:outline-none focus-within:border-teal-light-contrast ${this.props.className}`}
+    };
+
+    return (
+        <div className={`flex ${props.reverse ? 'flex-row-reverse' : 'flex-row'}`}>
+            <div className={`text-lg flex flex-grow m-2.5 items-center ${props.reverse ? 'justify-start' : 'justify-end'}`}>{props.label}</div>
+            <div className="flex items-center cursor-pointer relative" onClick={() => setShowDropdown(!showDropdown)}>
```

Comment:
Hm. @ErickSharp as we discussed on Discord you can simply name the function `handleClick`. I would not inline it, harder to read IMO.